propertyestimator/protocols/forcefield.py

```diff
@@ -635,3 +638,236 @@ def execute(self, directory, available_resources):
         logging.info(f'System generated: {self.id}')
 
         return self._get_output_dictionary()
+
+
+@register_calculation_protocol()
+class BuildTLeapSystem(BaseBuildSystemProtocol):
+    """Parametrise a set of molecules with an Amber based force field.
+    using the `tleap package <http://ambermd.org/AmberTools.php>`_.
+
+    Notes
+    -----
+    This protocol is currently a work in progress and as such has limited
+    functionality compared to the more established `BuildSmirnoffSystem` protocol.
+    """
+
+    @protocol_input(str)
+    def force_field_path(self, value):
+        """The file path to the force field parameters to assign to the system.
+        This path **must** point to a json serialized `TLeapForceFieldSource` object."""
+        pass
+
+    def __init__(self, protocol_id):
+        """Constructs a new `BuildTLeapSystem` object.
+        """
+        super().__init__(protocol_id)
+
```

Comment:
Jeff will add `create_mol2_from_topology_molecule` here

Reply:
The toolkit should assign the charges (with a choice between OETK and RDK/AmberTools) before writing out the mol2.

We should generate a conformer before writing out the topology molecule as well.

The topology will be created from a PDB file, so preserving the indexing is important.

Reply:
Added prototype of this functionality in https://github.com/openforcefield/propertyestimator/pull/99/commits/05f95e946a647a9b66228f96c81e9b1d12bb70c0

Reply:
Thanks @j-wags, this looks great!